### ISSUE TITLE
DOC/DEV: document env setup to test alternative backends locally

### ISSUE THREAD

**Issue by lucascolley** [scipy/scipy#21291]
### Issue with current documentation:

In https://github.com/scipy/scipy/issues/18867#issuecomment-2071546690 @dschmitz89 suggested that we should add some documentation to help devs set up an environment to test with alternative backends locally. While comprehensive documentation for the full range of operating systems and package managers people use seems infeasible, here are some notes on what works for me on Ubuntu with `mamba`.

### Idea or request for content:

Operating System: Ubuntu 22.04.4 LTS
GPU: NVIDIA GeForce GTX 1060 6GB

1. via `Software & Updates` -> `Additional Drivers`, select `Using NVIDIA driver metapackage from nvidia-driver-535 (proprietary, tested)` to install nvidia's graphics driver
2. Navigate to the directory of your SciPy clone. `nano environment.yml`:
	- Change `name` to something distinct e.g. `scipy-dev-xp`
	- Add the alternative backends you wish to test with under `dependencies`
		- WARNING: if you are including `jax`, you should specify `jax=0.4.28` (https://github.com/conda-forge/jaxlib-feedstock/issues/277)
		- WARNING: if you are including `pytorch`, you should remove the `openblas` and `libblas=*=*openblas  # helps avoid pulling in MKL` lines, as `pytorch` will pull in `mkl`.
3. `mamba env create -f environment.yml`
4. `mamba activate scipy-dev-xp`
5. `python dev.py build`
	- if you have included `pytorch`, instead run `python dev.py build -C-Dblas=blas -C-Dlapack=lapack -C-Duse-g77-abi=true`
6. Undo your changes to `environment.yml`
7. `SCIPY_DEVICE=cuda python dev.py test -b all`

**Comment by lucascolley** [comment]
I can confirm that the instructions in the top post now work for me.

**Comment by dschmitz89** [comment]
Could we add an additional environment yaml file for each alternative backend? Leaves pip users outside the loop though.

**Comment by lucascolley** [comment]
Ralf suggested in gh-21287 that we could take a look at `pixi` to generate lockfiles per platform and set of additional backends. I had a play around with it and it looks cool, although it will be quite complex given the amount of dependencies we have and the variation for BLAS etc.

**Comment by jakirkham** [comment]
> Ralf suggested in [gh-21287](https://github.com/scipy/scipy/issues/21287) that we could take a look at `pixi` to generate lockfiles per platform and set of additional backends. I had a play around with it and it looks cool, although it will be quite complex given the amount of dependencies we have and the variation for BLAS etc.

cc @wolfv (if there is someone interested on your end on helping Lucas get started with pixi 😉)

**Comment by rgommers** [comment]
> I had a play around with it and it looks cool, although it will be quite complex given the amount of dependencies we have and the variation for BLAS etc.

Yep:) Looks like I'm getting it to do what I want though: https://github.com/rgommers/pixi-dev-scipystack. It's not complete, but enough to get going with on array API library testing (commands like `test-jax`,  `test-torch`,`test-cpu` work on macOS, haven't tested on Linux). Having cross-platform lock files to bring a bit of method to the madness will be useful here. I plan on using this as my main dev setup locally for a while, adding what I need.

**Comment by lucascolley** [comment]
https://github.com/rgommers/pixi-dev-scipystack has largely addressed this, we can think about advertising it more broadly or integrating into the repo at some point down the line.
